src/core/public/application/types.ts

```diff
@@ -681,11 +703,9 @@ export interface ApplicationStart {
    * Navigate to a given app
    *
    * @param appId
-   * @param options.path - optional path inside application to deep link to.
-   *                       If undefined, will use {@link AppBase.defaultPath | the app's default path}` as default.
-   * @param options.state - optional state to forward to the application
+   * @param options - navigation options
    */
-  navigateToApp(appId: string, options?: { path?: string; state?: any }): Promise<void>;
+  navigateToApp(appId: string, options?: NavigateToAppOptions): Promise<void>;
```

Comment:
Should I also add the option to `navigateToUrl` ?

Reply:
We might need to do it for the sake of consistency. However, we can wait until someone requests this functionality.